tests/main/test_state.py

```diff
@@ -41,6 +42,7 @@ def capture_output(fn):
     return out.getvalue(), err.getvalue()
 
 
+@pytest.mark.skipif(sys.platform == 'win32', reason="Does not run on Windows, Python's issue")
```

Comment:
This is not a solution. Can we please make it pass? This will cause issues when using the library for Windows users.

Reply:
I find out why those cases fail.

In certain situations, we utilize tempfile.TemporaryDirectory() to establish a temporary directory that will be automatically deleted once our testing is complete. However, during the testing process, we generate files like *.db which lead to the creation of `file descriptions`. As a result, when we finish the test, the temp directory is deleted by the `tempfile` package, but the `file descriptions remain`, preventing the deletion of the directory. This occurs because Windows prohibits the deletion of a folder with already-used files.

And why it can be passed on to Linux instead of Windows?
On Linux and other Unix-like systems, you can usually delete an open file or its parent directory. This is because, on these systems, the file's data persists (and can be accessed by the process that has opened it) until all references to it are closed.

Because I think this isn't a problem with Ablator, just about the test case, so I directly skip it.
I will find a way to solve it.